File: VehicleController.js
#pragma strict

// rigidBody of the vehicle
public var Rb : Rigidbody;

// Speed of Movement in units per sec
private var speed : float = 4.5;

// For movement 
public var pos : Vector3;

// indicate if the player is inside this vehicle or not
public var inVehicle : boolean = false;

// For determining when/where to block and sprite rotation
public var isRight : boolean = false;
public var isLeft : boolean = false;
public var isUp : boolean = false;
public var isDown : boolean = false;

// Object that has the SpriteRender attached to it
public var sRendererObj : Transform;

// Sprite Renderer (Renders/displays the sprites)
public var sRenderer : SpriteRenderer;

// For switching the Sprite (Vechiles's direction) each time a different key is pressed
public var spriteLeft : Sprite;
public var spriteRight : Sprite;
public var spriteUp : Sprite;
public var spriteDown : Sprite;

// Used for pre-initialization
function Awake () 
{

	/*if (transform.CompareTag("Cement"))
	{
	
		Debug.Log("YOU LIKE ANUS!!!!");
		
		isCementTruck = true;
	
	}
	else if (transform.CompareTag("Manure"))
	{
	
		isManureTruck = true;
	
	}
	else
	{
	
		isCementTruck = false;
		isManureTruck = false;
	
	}*/
	
	// Intialize and set a reference to the Sprite Renderer that is attached to the same object as this script
	sRenderer = sRendererObj.GetComponent (SpriteRenderer);
	
	// intializes the RigidBody variable that is attached to Vehicle (player)
	Rb = transform.GetComponent(Rigidbody);

}

// use this for initialization
function Start () 
{

	// Take the initial position
	pos = transform.position;
	
	// if (tag == "DeLorean")
	// IF THIS VEHICLE IS A DELOREAN THEN
	// we can move it (should start inside as well as once we enter)
	// inVehicle = true

}

// called once per frame
function Update () 
{

	// The step size is equal to speed times frame time.
	var step = speed * Time.deltaTime;

	// move right if able (canMove == true) and key is pressed
	if (Input.GetKey (KeyCode.RightArrow) && inVehicle)
	{
		
		// Shorthand for writing Vector3(1, 0, 0).
		pos = Vector3.right;
		
		// move DeLorean (or vehicle whatever)
		transform.Translate (pos * step);
	
	}
	// move left if able (canMove == true) and key is pressed
	else if (Input.GetKey(KeyCode.LeftArrow) && inVehicle)
	{
	
		// Shorthand for writing Vector3(-1, 0, 0).
		pos = Vector3.left;
		
		transform.Translate (pos * step);
	
	}
	// move up if able (canMove == true) and key is pressed
	else if (Input.GetKey (KeyCode.UpArrow) && inVehicle)
	{
	
		// Shorthand for writing Vector3(0, 1, 0).
		pos = Vector3.up;
			
		transform.Translate (pos * step);
	
	}
	// move down if able (canMove == true) and key is pressed
	else if (Input.GetKey (KeyCode.DownArrow) && inVehicle)
	{
	
		// Shorthand for writing Vector3(0, -1, 0).
		pos = Vector3.down;
		
		transform.Translate (pos * step);
		
	}

}

function OnTriggerEnter (other : Collider)
{

	// vehicle is colliding with player
	if (other.tag == "Player")
	{
	
		// the player is in the vehicle
		//inVehicle = true;
	
	}

}

function OnTriggerExit (other : Collider)
{

	// vehicle is no longer colliding with player
	if (other.tag == "Player")
	{
	
		//inVehicle = false;
	
	}

}

File: _LEGACY/Back 2 the Futur3 Part 1 0.50/Assets/Standard Assets/Scripts/VehicleController.cs
using UnityEngine;
using System.Collections;

public class VehicleController : MonoBehaviour {

	// DeLorean as a GameObject to get scripts, positions, etc
	public GameObject theDeLorean;

	// Rigidbody of the DeLorean (this object)
	public Rigidbody rb;

	// x and y translations (for speed on x and y axis)
	public float xTranslation;
	public float yTranslation; 
	
	public bool isBlocked = false;
	
	public bool isLeft = false;
	public bool isRight = false;
	public bool isUp = false;
	public bool isDown = false;

	// mmmm..... thrust it
	public float thrust;

	// speed of DeLorean in Miles Per Hour
	public float deLoreanSpeed; 

	// stores actual velocity of the rigidbody attached to DeLorean GameObject
	//public Vector3  deLoreanVelocity;

	// Use this BEFORE the initialization
	void Awake ()
	{

		// Set variable "rb" to be the rigidBody attached to DeLorean GameObject
		rb = GetComponent<Rigidbody> ();

		thrust = 4;

	}

	// Use this for initialization
	void Start () 
	{
	


	}
	
	// Update is called once per frame
	void Update () 
	{
		
		//////////////////// GET KEY UP
		
		// when player releases a key
		if (Input.GetKeyUp (KeyCode.A))
		{
			
			// set "bool" to false because key is released
			isLeft = false;
			
		}
		else if (Input.GetKeyUp (KeyCode.D))
		{
			
			// set "bool" to false because key is released
			isRight = false;
			
		}
		else if (Input.GetKeyUp (KeyCode.W))
		{
			
			// set "bool" to false because key is released
			isUp = false; 
			
		}
		else if (Input.GetKeyUp (KeyCode.S))
		{
			
			// set "bool" to false because key is released
			isDown = false;
			
		} 

	}

	void FixedUpdate ()
	{

		// Always have it so the variable "deLoreanVelocity" is the actual current velocity of the RigidBody
		//deLoreanVelocity = rb.velocity;
		
		// 1 - Slow as shit
		// 2 - yupe still pretty slow
		// 3 - I guess thats ok
		// 4 - fastest you can go
		
		//xTranslation = Time.deltaTime * 2;
		//yTranslation = Time.deltaTime * 2;
		
		/*if (Input.GetKey (KeyCode.Space))
		{

			deLoreanVelocity = rb.velocity;

			Debug.Log ("velocity :" + deLoreanVelocity);
			Debug.Log ("velocity :" + rb.velocity);

		}*/
		
		// Go Leftward (when player presses key down like rapid fire)			
		if (Input.GetKey (KeyCode.A)) 
		{
			
			// Set bool "isLeft" to true
			isLeft = true;
			
			if (isBlocked == false)
			{
				
				rb.AddForce(new Vector3 (-2, 0, 0) * thrust);
				
				//velocity = new Vector3 (-100, 0, 0);
				//transform.Translate (rb.velocity * Time.deltaTime);
				
			}
			
		}
		// Go Rightward
		else if (Input.GetKey (KeyCode.D))
		{
			
			// Set bool "isRight" to true
			isRight = true;
			
			if (isBlocked == false)
			{
				
				rb.AddForce(new Vector3 (2, 0, 0) * thrust);
				
				//velocity = new Vector3 (100, 0, 0);
				//transform.Translate (rb.velocity * Time.deltaTime);
				
			}
			
		}
		// Go Up
		else if (Input.GetKey (KeyCode.W))
		{
			
			// Set bool "isUp" to true
			isUp = true;
			
			if (isBlocked == false)
			{
				
				rb.AddForce(new Vector3 (0, 0, 2) * thrust);
				
				//velocity = new Vector3 (0, 0, 100);
				//transform.Translate (rb.velocity * Time.deltaTime);
				
			}
			
		}
		// Go Backward (down)
		else if (Input.GetKey (KeyCode.S)) 
		{
			
			// set bool "isDown" to true
			isDown = true;
			
			if (isBlocked == false)
			{
				
				rb.AddForce(new Vector3 (0, 0, -2) * thrust);
				
				//velocity = new Vector3 (0, 0, -100);
				//transform.Translate (rb.velocity * Time.deltaTime);
				
			}
			
		}

	}

}
